### PR TITLE
Filtering for the COT admin list table.

### LINKED ISSUE #33196: [COT] Admin List > Implement Filters
Building on the work [done here](https://github.com/woocommerce/woocommerce/issues/32385), let's implement filters for the COT-powered admin list table.

This should mimic the functionality of the existing order list table as closely as reasonably possible.

### PR DESCRIPTION
**Pull request by barryhughes**
Implements filtering (by date, by customer) for the COT admin list table.

month-customer-filtering-cot

Closes #33196.

### How to test the changes in this Pull Request:

Setup:

1. Enable the COT feature.
2. Select **Use the WooCommerce orders tables** from **WooCommerce ▸ Settings ▸ Advanced ▸ Custom Data Stores**.
3. Ensure you have a few pages of order data covering different months and order statuses (tools like WC Smooth Generator can help create sufficient test data).

Testing:

1. Try filtering by month.
    - You should be able to reduce the result set to a single month.
    - You should be able to clear the month filter.
2. Try filtering by customer.
    - You should be able to reduce the result set to those orders placed by a single user.
    - You should be able to clear the customer filter.
    - You should be able to combine the customer filter with the month filter.
3. Ensure status changes via bulk actions still work.
    - Try filtering down to a single month and apply a bulk status change; the filter should remain in effect.

Notes:

- There appears to be a problem with view counts (ie, `All (1) | Completed (1) | ...`) but I believe this is unrelated to this work and should be addressed separately.

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by jorgeatorres** [comment]
Just mentioning that I've marked #33787 as ready for review, as the remaining changes were all in the `OrdersTableQuery` class. Hopefully that'll be merged soon and we can switch the base branch for this PR to `trunk`.

**Comment by barryhughes** [comment]
It makes sense to wait on https://github.com/woocommerce/woocommerce/pull/34119 before moving this one forward (as date filtering depends on this).

File: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
 use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+use Automattic\WooCommerce\RestApi\UnitTests\Helpers\ProductHelper;
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php]: This import doesn't seem to be required.

File: plugins/woocommerce/src/Internal/Admin/Orders/PageController.php [hunk 2]
```diff
@@ -63,8 +63,6 @@ private function verify_create_permission() {
 	}
 
 	/**
-=======
->>>>>>> trunk
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/Admin/Orders/PageController.php:67]: Git conflicts for the win! 😅

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hi @barryhughes!

This looks good to me 👍. I left some comments about a couple of tiny issues I found in the code, but other than that, this both looks good and works correctly.

Let me know what you think.

Thanks!

File: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php (hunk 1 above)
Comment (barryhughes) [reply plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php]: Good catch, thank you! Addressed in [`37151703`](https://github.com/woocommerce/woocommerce/pull/33789/commits/37151703714a16ab95526c51e048256e264b799e).

File: plugins/woocommerce/src/Internal/Admin/Orders/PageController.php (hunk 2 above)
Comment (barryhughes) [reply plugins/woocommerce/src/Internal/Admin/Orders/PageController.php:67]: Yes ... not sure how those snuck through ... ¯\\\_(ツ)_/¯

**Review by jorgeatorres — APPROVED** [review]
LGTM 🚢. Thank you @barryhughes!

**Pull request merged**